libsignal-service/src/provisioning/manager.rs

```diff
@@ -264,23 +261,16 @@ impl<P: PushService> ProvisioningManager<P> {
 
 #[derive(Clone)]
 pub struct LinkingManager<P: PushService> {
-    cfg: ServiceConfiguration,
-    user_agent: String,
-    password: String,
     push_service: P,
+    // used in credentials forwarded to the `ProvisioningManager`
+    password: Arc<String>,
```

Comment:
What is the benefit of Arc'ing this?

Reply:
Optimization for light-weight cloning.

Reply:
Sounds like premature optimization to me. `Arc` comes with its own overhead on every access, compared to a single malloc on clone (and clone happens rarely). Wouldn't do any of this without a clear indication for a bottleneck of some sort, and if there is a real bottleneck, I would use `password: &'a str` or `password: impl AsRef<str>` instead.

This just reads like poorly Java-ported code.

Reply:
I'll remove it for now and come back with a better argument when optimizing for clones (which happens on each incoming message).